### PR TITLE
nodel_labels

### PR DESCRIPTION
**Pull request by noragak**
- more than 8 groups
- node_label color
- group_labels
- group_label_colors

### CODE REVIEW

**Comment by noragak** [comment]
PR are always a bit difficult because my original project is at gitLab. I haven't found the best way to keep that clean.

**Comment by ericmjl** [comment]
No worries! We can work through it :smile:.

**Comment by noragak** [comment]
So what's next?

**Comment by ericmjl** [comment]
Let's take the chat to gitter: https://gitter.im/nxviz/Lobby#

File: nxviz/geometry.py [hunk 1]
```diff
@@ -24,6 +24,30 @@ def node_theta(nodelist, node):
...
 
 
+def group_theta(node_length, node_idx):
+
+    theta = -np.pi + node_idx*2*np.pi/node_length
+    return theta
+
+
+def text_alignement(x, y):
+
```
Comment (ericmjl) [inline nxviz/geometry.py]: This is definitely a great function to have! Let me take a stab at the necessary docstring below, could you modify it and add it to the function please?

```python

def text_alignment(x, y):
    """
    Align text labels based on the x- and y-axis coordinate values.

    This function is used for computing the appropriate alignment of the text 
    label. 

    For example, if the text is on the "right" side of the plot, we want it to 
    be left-aligned. If the text is on the "top" side of the plot, we want it
    to be bottom-aligned. 

    :param x, y: (`int` or `float`) x- and y-axis coordinate respectively.
    :returns: A 2-tuple of strings, the horizontal and vertical alignments
        respectively.
    """

File: nxviz/geometry.py [hunk 2]
```diff
@@ -24,6 +24,30 @@ def node_theta(nodelist, node):
     return theta
 
 
+def group_theta(node_length, node_idx):
+
```
Comment (ericmjl) [inline nxviz/geometry.py]: I'm having a little bit of trouble figuring out what exactly to write for the docstring. Could you take a stab at this one please?

```python
def group_theta(node_length, node_idx):
    """
    Returns an angle corresponding to the node index.

    Intended to be used for placing node group labels at the correct spot.

    :param float node_length: ...
    :param int node_idx: ...
    :returns: ...
    """

File: nxviz/plots.py [hunk 3]
```diff
@@ -201,18 +223,27 @@ def compute_node_colors(self):
                                             )
             self.sm._A = []
 
+    def compute_group_colors(self):
+        """Computes the group colors according to node colors
+        """
```
Comment (ericmjl) [inline nxviz/plots.py]: Minor nitpick: could you place the closing triple quote on the same line as the opening triple quote?

File: nxviz/utils.py [hunk 4]
```diff
@@ -121,6 +123,25 @@ def num_discrete_groups(data_container):
     return len(set(data_container))
 
 
+def itemes_in_groups(data_container):
```
Comment (ericmjl) [inline nxviz/utils.py]: I think the function name might be misspelled here - should probably be `items_in_groups`?

File: nxviz/geometry.py (hunk 1 above)
Comment (noragak) [reply nxviz/geometry.py]: Ah and alignment is misspelled!

**Comment by ericmjl** [comment]
Wonderful, @noragak! I'm happy to merge first. I'll add in a few tests for the new functions this weekend.

**Comment by noragak** [comment]
Nice! :)

**Pull request merged**